perf/benchmarkapps/GrpcClient/Program.cs

```diff
@@ -103,11 +108,45 @@ public static async Task<int> Main(string[] args)
                 await StartScenario();
 
                 await StopJobAsync();
-            });
+            }, new ReflectionBinder<ClientOptions>(options));
+
+            Log("gRPC Client");
 
             return await rootCommand.InvokeAsync(args);
         }
 
+        private class ReflectionBinder<T> : BinderBase<T> where T : new()
+        {
+            private readonly List<Option> _options;
+
+            public ReflectionBinder(List<Option> options)
+            {
+                _options = options;
+            }
+
+            protected override T GetBoundValue(BindingContext bindingContext)
+            {
+                var boundValue = new T();
+
+                Log($"Binding {typeof(T)}");
+
+                foreach (var option in _options)
+                {
+                    var value = bindingContext.ParseResult.GetValueForOption(option);
+
+                    var propertyInfo = typeof(T).GetProperty(option.Name, BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance);
```

Comment:
We own the `ClientOptions` type correct? Is using reflection the only way to assign the parsed values to the `ClientOptions` object here?

Reply:
We could store each option, then set each property individually using the option. But that would be a real pain in the butt.

It would look like this:
```cs
using System.CommandLine;
using System.CommandLine.Binding;

public class Program
{
    internal static async Task Main(string[] args)
    {
        var fileOption = new Option<FileInfo?>(
              name: "--file",
              description: "An option whose argument is parsed as a FileInfo",
              getDefaultValue: () => new FileInfo("scl.runtimeconfig.json"));

        var firstNameOption = new Option<string>(
              name: "--first-name",
              description: "Person.FirstName");

        var lastNameOption = new Option<string>(
              name: "--last-name",
              description: "Person.FirstName");

        var rootCommand = new RootCommand();
        rootCommand.Add(fileOption);
        rootCommand.Add(firstNameOption);
        rootCommand.Add(lastNameOption);

        rootCommand.SetHandler((FileInfo fileOptionValue, Person person) =>
        {
            DoRootCommand(fileOptionValue, person);
        },
        fileOption, new PersonBinder(firstNameOption, lastNameOption));

        await rootCommand.InvokeAsync(args);
    }

    public static void DoRootCommand(FileInfo aFile, Person aPerson)
    {
        Console.WriteLine($"File = {aFile?.FullName}");
        Console.WriteLine($"Person = {aPerson?.FirstName} {aPerson?.LastName}");
    }

    public class Person
    {
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
    }

    public class PersonBinder : BinderBase<Person>
    {
        private readonly Option<string> _firstNameOption;
        private readonly Option<string> _lastNameOption;

        public PersonBinder(Option<string> firstNameOption, Option<string> lastNameOption)
        {
            _firstNameOption = firstNameOption;
            _lastNameOption = lastNameOption;
        }

        protected override Person GetBoundValue(BindingContext bindingContext) =>
            new Person
            {
                FirstName = bindingContext.ParseResult.GetValueForOption(_firstNameOption),
                LastName = bindingContext.ParseResult.GetValueForOption(_lastNameOption)
            };
    }
}
```
But much longer because there are many more options. I think System.Commandline is going to improve it in the future with source gen. But the current state is way too verbose.

https://docs.microsoft.com/en-us/dotnet/standard/commandline/model-binding#model-binding-more-than-16-options-and-arguments